### ISSUE TITLE
Some Questions About the Model

### ISSUE THREAD

**Issue by manhcuong17072002** [dangtr0408/StyleTTS2-lite#5]
Hello, thank you for sharing this model. However, I have a few questions as follows:

- I see that you described this as a StyleTTS2-based model with some layers removed to make it lighter. Could you please explain the rationale behind removing those layers? Does this reduction affect the performance compared to the original StyleTTS2?

- If I don't want to remove any layers, or even want to increase the number of layers (as I prioritize quality over speed), is that possible? Would it produce better results?

Thank you for your contribution!

**Comment by manhcuong17072002** [comment]
@dangtr0408 Thank you for your feedback.

I would like to ask for some additional information as follows:
- How many hours of data did you train with and for how long?
- I have noticed that models like xtts2, f5tts, etc., which have been fine-tuned for Vietnamese, are not perfect or suitable for my purposes, and they require a very large amount of data that I cannot process entirely. Therefore, I would like to try lighter models as you have announced.

Thank you.

**Comment by dangtr0408** [comment]
Of course it is possible, for example you can totally swap the text encoder in this model to a bert (plbert, xphonebert,...). Visit the plbert repo, the authors show how to replace text encoder with plbert in styletts1 there, the process is pretty much the same with styletts2-lite.

The dataset is a subset of viVoice I didn't measure the total length but it is about 1/3 the size of viVoice in Gigabytes so probably around ~300 hours. I did 120k steps which is like ~1.5 epoch.

Lastly, i kept the dataset as it is, no addition modifications. But yes, I did notice the model suffers from inaccurate labeling.

**Comment by duynt575** [comment]
@dangtr0408 Is it possible to finetune this model with 12GB of VRAM? If it is I want to give it a go. I can finetune f5-tts and xtts v2 with 12GB of VRAM although xtts v2 is quite slow and the result is way worse than viXTTS after 4 epochs (224 hour dataset). f5-tts gives much better result for me. The appeal of this model is that it's very fast even with my old 6GB GPU.

**Comment by dangtr0408** [comment]
It is possible.
```
batch_size: 2
max_len: 120
```
It should use around 11 GB of VRAM with this setting. However i have seen feedback from the StyleTTS2 repo/discord server that setting max_len lower than 300 may hurt the model quality, personally i find the impact negligible.

**Comment by duynt575** [comment]
Thank you, I will try it then, need to figure out how to convert f5-tts dataset to StyleTTS format first.

**Comment by phvaha1** [comment]
@dangtr0408 @duynt575 
Do I need to convert the VIVOS dataset to phonemes before training the model?
for example the current vivoice dataset is: `xxx.wav | text1`
converted one is: `xxx.wav | phoneme1`

**Comment by phvaha1** [comment]
@manhcuong17072002 
Did you find out how to create dataset?

**Comment by manhcuong17072002** [comment]
@phvaha1 

Hello, I do not have the resources to train this model at the moment. I found out that it requires a large amount of data, similar to F5TTS, while the Vietnamese dataset from Vivoice is currently of poor quality and lacks accuracy—especially for sentences containing numbers. Therefore, I have decided to stop experimenting with this model. Thank you.
If you want to convert text to phonemes, you can use the library I am developing at:
🔗[Text2PhonemeFast](https://github.com/manhcuong02/Text2PhonemeFast)

However, this library is primarily designed for use with the XPhoneBERT model that I am currently working with:
🔗 https://github.com/VinAIResearch/XPhoneBERT
So it might not be suitable for this model. But you can give it a try and see how it works.

Thank you.

**Comment by phvaha1** [comment]
@manhcuong17072002 
thanks you for sharing, I will try it.

**Comment by dangtr0408** [comment]
Hello @phvaha1 , sorry for the late reply. Yes, you need to convert to phonemes before training. The pretrained Vietnamese model [uses Espeak](https://github.com/dangtr0408/StyleTTS2-lite/blob/db142897244e50623daa596de881a5fbbc486573/Demo/Space/app.py#L11-L28). 

You can definitely use a different G2P, but make sure that the G2P phoneme symbols matches the symbol set defined in [meldataset.py](https://github.com/dangtr0408/StyleTTS2-lite/blob/db142897244e50623daa596de881a5fbbc486573/meldataset.py#L21-L36). Please follow the instructions if you need to expand the symbol set (step 4).

**Comment by phvaha1** [comment]
@dangtr0408 
thanks you for reply, I will give it a try.

**Comment by phvaha1** [comment]
@dangtr0408 

> _extend = "∫̆ăη͡123456"

Could you explain where the _extend symbol above comes from?
eSpeak NG already supports Vietnamese, so why do we still need to extend the symbols?

**Comment by dangtr0408** [comment]
@phvaha1 espeak is not part of the model, it is a separate utilities we use to convert from texts to phonemes and for our model to understand we have to add the phonemes to its "dictionary". Originally the pretrained model by authors only supports characters/symbols for English espeak, not Vietnamese. You can see i added numbers 123456 which represent vietnamese tone marks in espeak.

**Comment by phvaha1** [comment]
@dangtr0408 
Thanks you, I got understand about `123456`, but what about `∫̆ăη͡`
Is it required?
